graphqlapi/src/main/java/io/stargate/graphql/schema/schemafirst/fetchers/admin/DeploySchemaFetcherBase.java

```diff
@@ -114,6 +114,7 @@
         dataStore.execute(query.build(dataStore)).get();
       }
       SchemaSource newSource = schemaSourceDao.insert(namespace, input);
+      schemaSourceDao.purgeOldSchemaEntries(namespace);
```

Comment:
This will be executed outside of the LWT transition (after the deployement_in_progress is set to `false`) but I don't think it is a problem. If another process (P2) will deploy a new schema before old entries are removed by P1 there will be a short period of time when there will be `NUMBER_OF_RETAINED_SCHEMA_VERSIONS+1` entries, but they will be purge by the P2 `purgeOldSchemaEntries` call.

There could be a situation that one process removes an entry that another tries to remove, but removing a non-existing entry is not problematic, right?

Reply:
Right, the delete queries target specific versions so they are idempotent.
A race condition is unlikely, and even then having an extra version for a short period of time would not be a problem.